pkg/oktaapi/client_wrapper.go

```diff
@@ -0,0 +1,174 @@
+		Email:      models.NewEmailAddress(o.Email),
+		EuaUserID:  o.Login,
+		// DisplayName: o.DisplayName,
+		// Email:       o.Email,
+		// Username:    o.Login,
+		// FirstName:   o.FirstName,
+		// LastName:    o.LastName,
+	}
+}
+
+// FetchUserInfos fetches users from Okta by EUA ID
+func (cw *ClientWrapper) FetchUserInfos(ctx context.Context, usernames []string) ([]*models.UserInfo, error) {
+	logger := appcontext.ZLogger(ctx)
+	users := []*models.UserInfo{}
+	if len(usernames) == 0 {
+		return users, nil
+	}
+
+	var euaSearches []string
+	for _, username := range usernames {
+		euaSearches = append(euaSearches, fmt.Sprintf(`profile.login eq "%v"`, username))
+	}
+	euaSearch := strings.Join(euaSearches, " or ")
+	searchString := fmt.Sprintf(`(%v)`, euaSearch)
+	search := query.NewQueryParams(query.WithSearch(searchString))
+
+	searchedUsers, _, err := cw.oktaClient.User.ListUsers(ctx, search)
+	if err != nil {
+		logger.Error("Error searching Okta users", zap.Error(err), zap.String("usernames", strings.Join(usernames, ", ")))
+		return nil, err
+	}
+
+	for _, user := range searchedUsers {
+		profile, err := cw.parseOktaProfileResponse(ctx, user.Profile)
+		if err != nil {
+			return nil, err
+		}
+		users = append(users, profile.toUserInfo())
+	}
+	return users, nil
+}
+
+// FetchUserInfo fetches a single user from Okta by EUA ID
+func (cw *ClientWrapper) FetchUserInfo(ctx context.Context, username string) (*models.UserInfo, error) {
+	logger := appcontext.ZLogger(ctx)
+
+	user, _, err := cw.oktaClient.User.GetUser(ctx, username)
+	if err != nil {
+		logger.Error("Error fetching Okta user", zap.Error(err), zap.String("username", username))
+		return nil, err
+	}
+
+	profile, err := cw.parseOktaProfileResponse(ctx, user.Profile)
+	if err != nil {
+		return nil, err
+	}
+
+	return profile.toUserInfo(), nil
+}
+
+// SearchCommonNameContains is a wrapper for SearchByName
+func (cw *ClientWrapper) SearchCommonNameContains(ctx context.Context, searchTerm string) ([]*models.UserInfo, error) {
+	return cw.SearchByName(ctx, searchTerm)
+}
+
+const euaSourceType = "EUA"
+const euaADSourceType = "EUA-AD"
+
+// SearchByName searches for a user by their First/Last name in Okta
+func (cw *ClientWrapper) SearchByName(ctx context.Context, searchTerm string) ([]*models.UserInfo, error) {
+	logger := appcontext.ZLogger(ctx)
+
+	// profile.SourceType can be EUA, EUA-AD, or cmsidm
+	// the first 2 represent EUA users, the latter represents users created directly in IDM
+	// status eq "ACTIVE" or status eq "STAGED" ensures we only get users who have EUAs (Staged means they just haven't logged in yet)
+	isFromEUA := fmt.Sprintf(`(profile.SourceType eq "%v" or profile.SourceType eq "%v")`, euaSourceType, euaADSourceType)
+	isActiveOrStaged := `(status eq "ACTIVE" or status eq "STAGED")`
+	nameSearch := fmt.Sprintf(`(profile.firstName sw "%v" or profile.lastName sw "%v" or profile.displayName sw "%v")`, searchTerm, searchTerm, searchTerm)
```

Comment:
We may want to document in a comment that this method only allows searching by the start/prefix of a user's first name/last name/display name; searching for "Dyl" will find "Dylan", but searching for "ylan" won't. This is due to [limitations in the Okta API](https://developer.okta.com/docs/reference/core-okta-api/#filter); it only supports the `sw` (starts with) operator, not `ew` (ends with), or any sort of wildcard.

Reply:
Addressed in [6420f51](https://github.com/CMSgov/easi-app/pull/2046/commits/6420f51a448742a7f3a31ce0745e6ab4428c5c78)